src/jit/lowerarmarch.cpp

```diff
@@ -63,107 +63,84 @@ bool Lowering::IsCallTargetInRange(void* addr)
 // Return Value:
 //    True if the immediate can be folded into an instruction,
 //    for example small enough and non-relocatable.
+//
+// TODO-CQ: we can contain a floating point 0.0 constant in a compare instruction
+// (vcmp on arm, fcmp on arm64).
+//
 bool Lowering::IsContainableImmed(GenTree* parentNode, GenTree* childNode)
 {
     if (varTypeIsFloating(parentNode->TypeGet()))
     {
-        // We can contain a floating point 0.0 constant in a compare instruction
-        switch (parentNode->OperGet())
-        {
-            default:
-                return false;
-
-            case GT_EQ:
-            case GT_NE:
-            case GT_LT:
-            case GT_LE:
-            case GT_GE:
-            case GT_GT:
-                if (childNode->IsIntegralConst(0))
-                {
-                    // TODO-ARM-Cleanup: not tested yet.
-                    NYI_ARM("ARM IsContainableImmed for floating point type");
-
-                    return true;
```

Comment:
@BruceForstall Why is this deleted for arm64?  

For arm 64 this could have been `OperIs(GT_EQ, GT_NE, GT_LT, GT_LE, GT_GE, GT_GT)`

Reply:
This is https://github.com/dotnet/coreclr/pull/17083. It doesn't actually get invoked. I changed it to an assert and never saw the assert hit. In a constructed example, the GT_EQ type came in as TYP_INT, not TYP_FLOAT/DOUBLE, so didn't go down this code path.

Reply:
I thought it looked familiar